pkg/validation/spec/schema.go

```diff
@@ -511,3 +516,33 @@ func (s *Schema) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+func (s *Schema) UnmarshalNextJSON(opts jsonv2.UnmarshalOptions, dec *jsonv2.Decoder) error {
+	var x struct {
+		Extensions
+		SchemaProps
+		SwaggerSchemaProps
+	}
+	if err := opts.UnmarshalNext(dec, &x); err != nil {
+		return err
+	}
+
+	_ = x.Ref.fromMap(x.Extensions)
+	_ = x.Schema.fromMap(x.Extensions)
+	delete(x.Extensions, "$ref")
+	delete(x.Extensions, "$schema")
+
+	// Necessary?
```

Comment:
resolve the question before merge?

Reply:
It is necessary as long as peer keys are included in the extensions as discussed in the other comment chain